nerfstudio/fields/nerfacto_field.py

```diff
@@ -239,17 +240,19 @@ def get_outputs(
         outputs_shape = ray_samples.frustums.directions.shape[:-1]
 
         # appearance
-        if self.training:
-            embedded_appearance = self.embedding_appearance(camera_indices)
-        else:
-            if self.use_average_appearance_embedding:
-                embedded_appearance = torch.ones(
-                    (*directions.shape[:-1], self.appearance_embedding_dim), device=directions.device
-                ) * self.embedding_appearance.mean(dim=0)
+        embedded_appearance = None
+        if self.appearance_embedding_dim > 0:
```

Comment:
(2) and then condition this on `if self.embedding_appearance is not None`

because it seems statically safer for `self.embedding_appearance` to be typed as `Embedding | None`